qiskit_experiments/calibration/calibration_definitions.py

```diff
@@ -0,0 +1,382 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class to store the results of a calibration experiments."""
+
+import copy
+from datetime import datetime
+import dataclasses
+from typing import Tuple, Union, List, Optional, Type
+import pandas as pd
+
+from qiskit.circuit import Gate
+from qiskit import QuantumCircuit
+from qiskit.pulse import Schedule, DriveChannel, ControlChannel, MeasureChannel
+from qiskit.pulse.channels import PulseChannel
+from qiskit.circuit import Parameter
+from .exceptions import CalibrationError
+from .parameter_value import ParameterValue
+
+
+class CalibrationsDefinition:
+    """
+    A class to manage schedules with calibrated parameter values.
+    Schedules are stored in a dict and are intended to be fully parameterized,
+    including the index of the channels. The parameters are therefore stored in
+    the schedules. The names of the parameters must be unique. The calibrated
+    values of the parameters are stored in a dictionary.
+    """
+
+    def __init__(self, backend):
+        """
+        Initialize the instructions from a given backend.
+
+        Args:
+            backend: The backend from which to get the configuration.
+        """
+
+        self._n_qubits = backend.configuration().num_qubits
+        self._n_uchannels = backend.configuration().n_uchannels
+        self._config = backend.configuration()
+        self._params = {'qubit_freq': {}}
```

Comment:
In our conventional database this has been organized as somewhat like `Dict[qubit(s)][pulse name][param name]`. Do you think this new data structure, i.e. `Dict[param name][channel]` is reasonable? @wshanks